README.md

```diff
@@ -98,6 +98,37 @@ subscript(index: Int) -> T {
 
 _Rationale:_ The intent and meaning of the first version is clear, and results in less code.
 
+#### Avoid redundant type declarations
+
+When type information can be inferred from an initial value, avoid declaring it explicitly, unless doing so is an active decision to improve code legibility.
+
+```swift
+var isOnFire = false
+var referenceFrame = CGRect(x: 0, y: 0, width: 100, height: 100)
+```
+
+instead of:
+
+```swift
+var isOnFire: Bool = false
+var referenceFrame: CGRect = CGRect(x: 0, y: 0, width: 100, height: 100)
+```
+
+When type information _cannot_ be inferred from an initial value then it should be explicitly added to the declaration; you should not add type information to a constructor for the purpose of allowing type inference.
```

Comment:
Ehhh. We can probably strip this bit out, not least because we're inconsistent about it—for example, using `Result<T, E>.Success(value)` as an initial value.